src/gui/qgsmapcanvas.h

```diff
@@ -786,9 +787,9 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
      */
     void mapToolSet( QgsMapTool *newTool, QgsMapTool *oldTool );
 
-    // ### QGIS 3: remove the signal
+
     //! Emitted when selection in any layer gets changed
-    void selectionChanged( QgsMapLayer *layer );
+    void selectionChanged( QgsVectorLayer *layer );
```

Comment:
I think this method has to go, it really has nothing to do with the map canvas ;)

Reply:
In fact, connect to the selectionChanged signal of the currentLayer only

Reply:
@m-kuhn I kind of like this top level one because you don't have to care about which layer selection was on, just that there was one.  Otherwise you need to connect to the current layer one and keep it in sync when that changes which is two signals vs just one.

Reply:
Now a setExtent is called even when nothing changes which is not very nice. If a generic selectionChanged signal is emitted somewhere I'd rather put it the layer registry.